src/steps/resource-manager/appservice/client.ts

```diff
@@ -37,4 +42,96 @@ export class AppServiceClient extends Client {
       callback,
     });
   }
+
+  public createAppAuthConfigurationContext(): (
+    name?: string,
+    resourceGroup?: string,
+  ) => Promise<WebAppsGetAuthSettingsResponse | undefined> {
+    let canGetAuthSettings: boolean | undefined = undefined;
+
+    return async (
+      name?: string,
+      resourceGroup?: string,
+    ): Promise<WebAppsGetAuthSettingsResponse | undefined> => {
+      if (typeof canGetAuthSettings === 'undefined') {
+        canGetAuthSettings = await this.canFetchAppAuthSettings(
+          name,
+          resourceGroup,
+        );
+      }
+
+      if (!canGetAuthSettings) {
+        return;
+      }
+
+      const serviceClient = await this.getAuthenticatedServiceClient(
+        WebSiteManagementClient,
+      );
+
+      if (!name || !resourceGroup) {
+        return;
+      }
+
+      return serviceClient.webApps.getAuthSettings(resourceGroup, name);
+    };
+  }
+
+  public async canFetchAppAuthSettings(
+    name: string | undefined,
+    resourceGroup: string | undefined,
+  ): Promise<boolean | undefined> {
+    const serviceClient = await this.getAuthenticatedServiceClient(
+      WebSiteManagementClient,
+    );
+
+    if (!name || !resourceGroup) {
+      return;
+    }
+
+    try {
+      await serviceClient.webApps.getAuthSettings(resourceGroup, name);
+      return true;
+    } catch (err) {
+      this.logger.warn({ err }, 'Warning: unable to fetch app configuration.');
+      if (err.statusCode === 403 && err.code === 'AuthorizationFailed') {
+        this.logger.publishEvent({
+          name: 'MISSING_PERMISSION',
```

Comment:
We use the convention of lowercase events.
```suggestion
          name: 'missing_permission',
```